### PR TITLE
Xournal: Darwin support

### PR DESCRIPTION
**Pull request by johbo**
###### Motivation for this change

This PR has two small changes:

1. It allows to build Xournal on Darwin.
2. In enables xinput also on Darwin, so that e.g. pressure sensitivity works in Xournal.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/graphics/xournal/default.nix [hunk 1]
```diff
@@ -1,8 +1,10 @@
-{ stdenv, fetchurl, makeDesktopItem
+{ stdenv, lib, fetchurl, makeDesktopItem
```
Comment (LnL7) [inline pkgs/applications/graphics/xournal/default.nix]: You can access lib using `stdenv.lib`

File: pkgs/applications/graphics/xournal/default.nix [hunk 2]
```diff
@@ -50,8 +54,8 @@ stdenv.mkDerivation rec {
   meta = {
```
Comment (LnL7) [inline pkgs/applications/graphics/xournal/default.nix]: You can use `meta = with stdenv.lib; ...` to get the attributes like licenses into scope here.

**Comment by johbo** [comment]
Updated the style. 

@LnL7 is using `lib` in the attributes at the head discouraged? I was not too sure about it, think I saw different styles in different files.

File: pkgs/applications/graphics/xournal/default.nix [hunk 3]
```diff
@@ -2,7 +2,9 @@
 , ghostscript, atk, gtk2, glib, fontconfig, freetype
 , libgnomecanvas, libgnomeprint, libgnomeprintui
 , pango, libX11, xproto, zlib, poppler
-, autoconf, automake, libtool, pkgconfig}:
```
Comment (mdaiter) [inline pkgs/applications/graphics/xournal/default.nix]: Why was this changed? Seems like it's the same package list, but the end-bracket's on a different line.

File: pkgs/applications/graphics/xournal/default.nix [hunk 4]
```diff
@@ -47,11 +51,11 @@ stdenv.mkDerivation rec {
...
-  meta = {
+  meta = with stdenv.lib; {
     homepage = http://xournal.sourceforge.net/;
     description = "Note-taking application (supposes stylus)";
-    maintainers = [ stdenv.lib.maintainers.guibert ];
-    license = stdenv.lib.licenses.gpl2;
-    platforms = stdenv.lib.platforms.linux;
+    maintainers = [ maintainers.guibert ];
+    license = licenses.gpl2;
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (mdaiter) [inline pkgs/applications/graphics/xournal/default.nix]: This should be written as `platforms = with platforms; [ linux darwin ];`

**Review by mdaiter — CHANGES_REQUESTED** [review]
Good, but a bit of touch-up requested

File: pkgs/applications/graphics/xournal/default.nix (hunk 4 above)
Comment (LnL7) [inline pkgs/applications/graphics/xournal/default.nix]: While nested lists should work on hydra, it's not considered to be a best practice.

**Comment by LnL7** [comment]
> is using `lib` in the attributes at the head discouraged? I was not too sure about it, think I saw different styles in different files.

Not really, but AFAIK `stdenv.lib` is used a lot more for package definitions, probably because a most packages already need stdenv for mkDerivation, ...

**Comment by lverns** [comment]
These changes seem to build and run fine for me on NixOS.

**Comment by johbo** [comment]
Updated the style as requested - hope it's ready to go now 😄

File: pkgs/applications/graphics/xournal/default.nix [hunk 5]
```diff
@@ -47,11 +49,11 @@ stdenv.mkDerivation rec {
...
-  meta = {
+  meta = with stdenv.lib; {
     homepage = http://xournal.sourceforge.net/;
     description = "Note-taking application (supposes stylus)";
-    maintainers = [ stdenv.lib.maintainers.guibert ];
-    license = stdenv.lib.licenses.gpl2;
-    platforms = stdenv.lib.platforms.linux;
+    maintainers = [ maintainers.guibert ];
+    license = licenses.gpl2;
+    platforms = with platforms; [ linux darwin ];
```
Comment (LnL7) [inline pkgs/applications/graphics/xournal/default.nix]: Attributes in platforms [are lists](https://github.com/NixOS/nixpkgs/blob/master/lib/platforms.nix) so this will result in nested lists: `[ [ "i686-linux" ... ] [ "x86_64-darwin" ] ]`

Comment (johbo) [inline pkgs/applications/graphics/xournal/default.nix]: LOL, got trapped by the other comments, I'll fix this back again

Comment (LnL7) [inline pkgs/applications/graphics/xournal/default.nix]: Sorry for the confusion, my previous comment wasn't very clear.

**Review by LnL7 — APPROVED** [review]
